packages/frontend-api/src/Model/Resolver/Image/ImagesResolver.php

```diff
@@ -0,0 +1,138 @@
+     */
+    protected $imageFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Image\Config\ImageConfig
+     */
+    protected $imageConfig;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Image\ImageFacade $imageFacade
+     * @param \Shopsys\FrameworkBundle\Component\Image\Config\ImageConfig $imageConfig
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        ImageFacade $imageFacade,
+        ImageConfig $imageConfig,
+        Domain $domain
+    ) {
+        $this->imageFacade = $imageFacade;
+        $this->imageConfig = $imageConfig;
+        $this->domain = $domain;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @param string|null $type
+     * @param string|null $size
+     * @return array
+     */
+    public function resolveByProduct(Product $product, ?string $type, ?string $size): array
+    {
+        return $this->resolveByEntity($product, static::IMAGE_PRODUCT_ENTITY, $type, $size);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Category\Category $category
+     * @param string|null $type
+     * @param string|null $size
+     * @return array
+     */
+    public function resolveByCategory(Category $category, ?string $type, ?string $size): array
+    {
+        return $this->resolveByEntity($category, static::IMAGE_CATEGORY_ENTITY, $type, $size);
+    }
+
+    /**
+     * @param object $entity
+     * @param string $entityName
+     * @param string|null $type
+     * @param string|null $size
+     * @return array
+     */
+    protected function resolveByEntity(object $entity, string $entityName, ?string $type, ?string $size): array
+    {
+        $resolvedImages = [];
+        $sizeConfigs = $this->getSizeConfigs($type, $size, $entityName);
+        $images = $this->imageFacade->getImagesByEntityIndexedById($entity, $type);
+
+        foreach ($images as $image) {
+            foreach ($sizeConfigs as $sizeConfig) {
+                $resolvedImages[] = [
+                    'type' => $image->getType(),
+                    'position' => $image->getPosition(),
+                    'width' => $sizeConfig->getWidth(),
+                    'height' => $sizeConfig->getHeight(),
+                    'size' => $sizeConfig->getName() === null ? ImageConfig::DEFAULT_SIZE_NAME : $sizeConfig->getName(),
+                    'url' => $this->imageFacade->getImageUrl($this->domain->getCurrentDomainConfig(), $image, $sizeConfig->getName(), $image->getType()),
+                ];
+            }
+        }
+
+        return $resolvedImages;
+    }
+
+    /**
+     * @param string|null $type
+     * @param string|null $size
+     * @param string $entityName
+     * @return array|\Shopsys\FrameworkBundle\Component\Image\Config\ImageSizeConfig[]
```

Comment:
Can it return array of something different than `\Shopsys\FrameworkBundle\Component\Image\Config\ImageSizeConfig`?

Reply:
No, fixed